src/wrapper/wrapper.c

```diff
@@ -249,13 +217,14 @@ real_call:;
 
 FILE *fopen (const char *path, const char *mode)
 {
-    if (ctx == NULL) {
-        dyad_sync_init ();
-    }
     char *error = NULL;
     typedef FILE *(*fopen_ptr_t) (const char *, const char *);
     fopen_ptr_t func_ptr = NULL;
 
+    if (ctx == NULL) {
+        dyad_wrapper_init ();
+    }
+
```

Comment:
If we're going to move this check after the variable initialization for `fopen`, then we should do the same for `open`.

Reply:
The same should probably be true for `close` and `fclose`. Currently, `fopen` is the only function that does not start with this check.

Reply:
Good point. I made similar changes to open, close, and fclose.